cmake/VersionString.cmake

```diff
@@ -202,7 +202,7 @@ function(parse_version_file names file)
 					set(newbuild "${CMAKE_MATCH_2}${CMAKE_MATCH_3}")
 				elseif(release MATCHES "^\\-rc([0-9]+)$")
 					set(prerelease 1)
-					set(newpatch "9999")
+					set(newpatch "255")
```

Comment:
I don't think changing the numbers here is the correct solution: On the one hand other platforms don't have this limitation (Windows has four 16-bit components, Linux stores strings) and `9999` makes it clearer that the version is special. On the other hand, version numbers > 255 can also come from other places such as the date based version numbers in development snapshots (latest snapshot version "1.2-dev-2016-12-19" is encoded as "1.1.2016.1219". Four components are also used for release candidates: "1.2-rc4" is encoded as "1.1.9999.4".

I'm not sure how to best fit this into the Mac OS X restrictions.
One solution could be to keep the version numbers as-is and then only limit the numbers when used for Mach-O binaries (losing information).